pandas/core/arrays/string_arrow.py

```diff
@@ -243,6 +155,10 @@ def _from_sequence(cls, scalars, dtype: Dtype | None = None, copy: bool = False)
 
         cls._chk_pyarrow_available()
 
+        if dtype and not (isinstance(dtype, str) and dtype == "string"):
```

Comment:
can we add some introspection functions

is_string_python_dtype
is_string_arrow_dtype
is_string_dtype

to encompass operations

Reply:
followon ok if you can create an issue

Reply:
can discuss this once we have merged what we got in case things change. This conditional is slightly different from what would be in the introspection functions since this allows "string" whatever the global storage whereas StringDtype(), without the storage given uses the global option.

I'm probably overthinking or completely misunderstood @jorisvandenbossche suggestion that "StringDtype()" should defer the lookup.